src/line.cpp

```diff
@@ -271,6 +273,23 @@ int manhattan_dist( const point &loc1, const point &loc2 )
     return d.x + d.y;
 }
 
+int octile_dist( const point &loc1, const point &loc2, int multiplier )
+{
+    const point d = ( loc1 - loc2 ).abs();
+    const int mind = std::min( d.x, d.y );
+    const int maxd = std::max( d.x, d.y );
+    // sqrt(2) is approximately 99 / 70
+    return mind * multiplier + ( maxd - mind ) * multiplier * 99 / 70;
+}
+
+float octile_dist_exact( const point &loc1, const point &loc2 )
+{
+    const point d = ( loc1 - loc2 ).abs();
+    const int mind = std::min( d.x, d.y );
+    const int maxd = std::max( d.x, d.y );
+    return mind + ( maxd - mind ) * sqrt2;
```

Comment:
```suggestion
    return dx + dy + ( sqrt2 - 2 ) * mind;
```
Avoids one branch. (<https://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html#heuristics-for-grid-maps>)

Reply:
std::min and std::max don't cause branching nowadays. Modern compilers targeting modern CPUs generate a conditional-move operation, which doesn't incur the overhead of branching.
I suppose it does save 1 line of code, hopefully without harming readability?
Anyway, done.